src/zenml/integrations/vault/secrets_manager/vault_secrets_manager.py

```diff
@@ -0,0 +1,333 @@
+from zenml.integrations.vault import VAULT_SECRETS_MANAGER_FLAVOR
+from zenml.logger import get_logger
+from zenml.secret.base_secret import BaseSecretSchema
+from zenml.secret.secret_schema_class_registry import SecretSchemaClassRegistry
+from zenml.secrets_managers.base_secrets_manager import BaseSecretsManager
+
+logger = get_logger(__name__)
+
+ZENML_SCHEMA_NAME = "zenml-schema-name"
+ZENML_PATH = "zenml"
+
+
+def sanitize_secret_name(secret_name: str) -> str:
+    """Sanitize the secret name to be used in Vault."""
+    return re.sub(r"[^0-9a-zA-Z_\.]+", "_", secret_name).strip("-_.")
+
+
+def prepend_secret_schema_to_secret_name(secret: BaseSecretSchema) -> str:
+    """
+    Prepend the secret schema name to the secret name.
+
+    Args:
+        secret: The secret to prepend the secret schema name to.
+
+    Returns:
+        The secret name with the secret schema name prepended.
+    """
+    secret_name = sanitize_secret_name(secret.name)
+    secret_schema_name = secret.TYPE
+    return f"{secret_schema_name}-{secret_name}"
+
+
+def remove_secret_schema_name(combined_secret_name: str) -> str:
+    """
+    Remove the secret schema name from the secret name.
+
+    Args:
+        combined_secret_name: The secret name to remove the secret schema name from.
+
+    Returns:
+        The secret name with the secret schema name removed.
+    """
+    if "-" in combined_secret_name:
+        return combined_secret_name.split("-")[1]
+    else:
+        raise RuntimeError(
+            f"Secret name `{combined_secret_name}` does not have a "
+            f"Secret_schema name on it."
+        )
+
+
+def get_secret_schema_name(combined_secret_name: str) -> str:
+    """
+    Get the secret schema name from the secret name.
+
+    Args:
+        combined_secret_name: The secret name to get the secret schema name from.
+
+    Returns:
+        The secret schema name.
+    """
+    if "-" in combined_secret_name:
+        return combined_secret_name.split("-")[0]
+    else:
+        raise RuntimeError(
+            f"Secret name `{combined_secret_name}` does not have a "
+            f"Secret_schema name on it."
+        )
+
+
+class VaultSecretsManager(BaseSecretsManager):
+    """Class to interact with the Vault secrets manager - Key/value Engine.
+
+    Attributes:
+        url: The url of the Vault server.
+        token: The token to use to authenticate with Vault.
+        cert: The path to the certificate to use to authenticate with Vault.
+        verify: Whether to verify the certificate or not.
+        mount_point: The mount point of the secrets manager.
+        namespace: The namespace of the secrets manager.
+    """
+
+    # Class configuration
+    FLAVOR: ClassVar[str] = VAULT_SECRETS_MANAGER_FLAVOR
+    CLIENT: ClassVar[Any] = None
+
+    url: str
+    token: str
+    cert: Optional[str]
+    verify: Optional[str]
+    namespace: Optional[str]
+    mount_point: Optional[str] = "secret"
```

Comment:
If I understood you correctly @safoinme , the secret mountpoint is the standard mount point that any hvac deployment will have/ autogenerate?

Reply:
Yes the secret moint_point is the standard path (mount point) that vault KV secrets engine deploy or create by default in every new deployment. @karimhabush maybe you could give more clarification on this.

Reply:
I agree with what you said @safoinme , by default, when the `mount_point` isn't given, `hvac` will look for secrets within the path `secret/`. However, it is important to note that Vault server itself only creates the path by default in a dev environment. And the `mount_point` argument should be specified when working on production.

Reply:
I agree with you @safoinme , however it is important to note that `secret` is the default expected mount point for hvac, however it does not generate it if it wasn't already setup on the Vault server.

Reply:
Then if there is a chance, that this mount point might not exist, I would make this field required and remove the default value. The user should consciously choose the mountpoint then.